web/js/codeworld.js

```diff
@@ -171,6 +171,9 @@ function initCodeworld() {
 
     window.codeworldEditor.on('changes', window.updateUI);
 
+    setTimeout(codeToDoc, 5000);
+    setInterval(codeToDoc, 30000);
```

Comment:
I wonder if you should do something like:

    window.reparseTimeoutId = null;
    window.codeworldEditor.on("change", () => {
      if (window.reparseTimeoutId) clearTimeout(window.reparseTimeoutId);
      window.reparseTimeoutId = setTimeout(codeToDoc, 3000);
    });

That way, the symbol list would be relatively up-to-date (max 3 seconds old), but you wouldn't waste time constantly reparsing when the document isn't modified, and you also wouldn't risk typing being choppy because of constant reparses while the user is typing.  I think that's the right trade-off here.